src/software/backend/robot_status.cpp

```diff
@@ -0,0 +1,101 @@
+#include "software/backend/robot_status.h"
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "software/logger/logger.h"
+
+std::unique_ptr<TbotsRobotMsg> convertRobotStatusToTbotsRobotMsg(
+    const RobotStatus& robot_status)
+{
+    //    std::map<std::string, ErrorCode> errorStringToEnumMap;
+    std::vector<std::string> errorStrings = {
+        "charge timeout",
+        "wheel 0 motor hot",
+        "wheel 1 motor hot",
+        "wheel 2 motor hot",
+        "wheel 3 motor hot",
+        "dribbler motor hot",
+        "wheel 0 encoder not commutating",
+        "wheel 1 encoder not commutating",
+        "wheel 2 encoder not commutating",
+        "wheel 3 encoder not commutating",
+        "wheel 0 Hall sensor stuck low",
+        "wheel 1 Hall sensor stuck low",
+        "wheel 2 Hall sensor stuck low",
+        "wheel 3 Hall sensor stuck low",
+        "dribbler Hall sensor stuck low",
+        "wheel 0 Hall sensor stuck high",
+        "wheel 1 Hall sensor stuck high",
+        "wheel 2 Hall sensor stuck high",
+        "wheel 3 Hall sensor stuck high",
+        "dribbler Hall sensor stuck high",
+    };
+
+    // Insufficient information to make the TbotsRobotMsg fields for
+    // ChipperKickerStatus, DriveUnits, and NetworkStatus
+    auto robot_msg = std::make_unique<TbotsRobotMsg>();
+
+    robot_msg->set_robot_id(robot_status.robot);
+
+    ErrorCode error_code;
+    for (auto& msg : robot_status.robot_messages)
+    {
+        // If it is a valid error message, add to error_code
+        if (std::find(errorStrings.begin(), errorStrings.end(), msg) !=
+            errorStrings.end())
+        {
```

Comment:
this check isn't necessary because we'll know whether an error string is valid from ErrorCode_Parse

Reply:
Done.